src/NotificationTarget.php

```diff
@@ -543,18 +556,23 @@ public function addToRecipientsList(array $data)
         if (isset($data['name']) && !empty($data['name'])) {
             $username = $data['name'];
         }
+
         if (isset($data['users_id']) && ($data['users_id'] > 0)) {
             $user = new User();
             if (
                 !$user->getFromDB($data['users_id'])
                 || ($user->getField('is_deleted') == 1)
+                // if notification not send by NotificationTargetCommonITILObject
+                // check if user explicitly refused notification
+                //for CommonITILObject use_notification is check before
+                || (!$this->validateSendTo("", $data, true, null, get_called_class()))
```

Comment:
Not passing a correct value for all parameters would produce unexpected results. I propose to rely on `validateSendTo()` check done when notification will be sent.
```suggestion
```

Reply:
done